client/src/comps/SingleVideo.jsx

```diff
@@ -0,0 +1,76 @@
+import React, { useState } from "react";
+import "../styles/SingleVideo.css";
+import ThumbUpIcon from "@mui/icons-material/ThumbUp";
+import ThumbDownIcon from "@mui/icons-material/ThumbDown";
+
+function SingleVideo(props) {
+  const [likes, setLikes] = useState(props.rating);
+
+  function handleLikes() {
+    setLikes((rating) => rating + 1);
+    updateVideoRating({ ...props, rating: likes + 1 });
+  }
+
+  function handleUnlikes() {
+    setLikes((rating) => rating - 1);
+    updateVideoRating({ ...props, rating: likes - 1 });
+  }
+
+  const updateVideoRating = async (updatedVideo) => {
+    fetch(
+      `https://full-stack-project-assessment-gb1q.onrender.com/${updatedVideo.id}`,
+      {
+        method: "PUT",
+        headers: {
+          "Content-Type": "application/json",
+        },
+        body: JSON.stringify({ rating: updatedVideo.rating }),
+      }
+    )
+      .then((res) => {
+        if (res.ok) {
+          console.log("Video rating updated");
+        } else {
+          throw new Error("Failed to update video rating");
```

Comment:
Well done for remembering this check, i always forget about this!

Reply:
An extension might be, could we show the user anything in this case if this were to happen? 😄